tools/data_exploration/graph_lcbd.r

```diff
@@ -0,0 +1,233 @@
+
+if (!is.na(lat)) {
+  data_xy <- data_full[, c("latitude", "longitude")]
+  # Data for environment
+  data_env <- data_full[, c("site", "latitude", "longitude")]
+}
+
+# Data with only species and their abundance
+data_spe <- data_spe[, -1]
+
+#####Your analysis
+
+# Computation beta.div {adespatial}
+# Beta.div on Hellinger-transformed species data
+data_beta <- adespatial::beta.div(data_spe, method = "hellinger", nperm = 9999)
+
+beta <- capture.output(summary(data_beta))
+write(beta, "LCBD.txt")
+
+# Which species have a SCBD larger than the mean SCBD?
+scbd <- capture.output(data_beta$SCBD[data_beta$SCBD >= mean(data_beta$SCBD)])
+write(scbd, "SCBD.txt")
+
+##1st fonction
+beta_div_ext <- function(data_beta, data_xy, data_env) {
+   data_beta_ext <- data.frame(data_xy, data_env, LCBD = data_beta$LCBD * 100, p.LCBD = data_beta$p.LCBD, signif = data_beta$p.LCBD < 0.05)
+
+  graph_beta_ext <- ggplot2::ggplot(data = data_beta_ext, ggplot2::aes(x = latitude, y = longitude, size = LCBD, col = signif)) +
+  ggplot2::geom_point() +
+  ggplot2::scale_colour_manual(values = c("grey56", "black"), labels = c("Non significant", "Significant"), name = "Significance at 0.05") +
+  ggplot2::xlab("Longitude") + ggplot2::ylab("Latitude")
+
+  ggplot2::ggsave("Beta_diversity_ext.png", graph_beta_ext)
+}
+
+## Boyé et al. 2017 JSR Fig R
+####################################################
+
+####LCBD####
+compute_lcbd <- function(data_spe, data_full) {
+  lcbd_site <- adespatial::beta.div(data_spe, "hellinger", nperm = 999)
+
+#############
+  mat_lcbd_site <- data.frame(data_spe, LCBD = lcbd_site$LCBD * 100, p.LCBD = lcbd_site$p.LCBD, signif = lcbd_site$p.LCBD < 0.05, site = data_full$site, date = data_full$date)
+
+## Map spatio-temp
+##################
+  p1 <- ggplot2::qplot(date, site, size = LCBD, col = signif, data = mat_lcbd_site)
+  p1 <- p1 + ggplot2::scale_colour_manual(values = c("grey56", "black"), labels = c("Non significant", "Significant"), name = "Significance at 0.05")
+  p1 <- p1 + ggplot2::theme_bw() + ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90)) + ggplot2::xlab("Date") + ggplot2::ylab("Site")
+
+  ggplot2::ggsave("Spatio_temp.png", p1)
+
+
+## Par années
+#############
+  mean_time <- tapply(mat_lcbd_site$LCBD, mat_lcbd_site$date, mean)
+  sd_time <- tapply(mat_lcbd_site$LCBD, mat_lcbd_site$date, sd)
+  year <- unique(mat_lcbd_site$date)
+
+  data <- data.frame(year, mean_time, sd_time)
+
+  time <- ggplot2::ggplot() + ggplot2::geom_pointrange(ggplot2::aes(x = year, y = mean_time, ymin = mean_time - sd_time, ymax = mean_time + sd_time), data = data)
+  time <- time + ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90), axis.line.y = ggplot2::element_line(size = 0.5)) + ggplot2::ylab("mean LCBD")
+
+  ggplot2::ggsave("Time.png", time)
+}
+
+## Choose another graph
+#######################
+compute_lcbd2 <- function(data_spe, data_full) {
+  lcbd_site <- adespatial::beta.div(data_spe, "hellinger", nperm = 999)
```

Comment:
Different number of permutations in this one, normal ?

Reply:
Maybe you could compute beta diversity only once to shorten the run time because it seems to take a really long time to run